airflow/settings.py

```diff
@@ -183,6 +183,22 @@ def configure_orm(disable_connection_pool=False):
     # For Python2 we get back a newstr and need a str
     engine_args['encoding'] = engine_args['encoding'].__str__()
 
+    # Sane timeouts for connections and statements
+    connect_args = {}
+
+    try:
+        connect_timeout_seconds = conf.getint('core', 'SQL_ALCHEMY_CONNECT_TIMEOUT_SECONDS')
+    except conf.AirflowConfigException:
+        connect_timeout_seconds = 30
+    connect_args['connect_timeout'] = connect_timeout_seconds
```

Comment:
Do we know if this works for all databases that are currently supported (SQLite, MySQL, Postgres)